File: Utilities/Stack.js
class Node {
  constructor(data, next = null) {
    this.data = data
    this.next = next
  }
}

class Stack {
  constructor(top = null) {
    this.top = top
  }

  push(data) {
    const currNode = new Node(data)
    currNode.next = this.top
    this.top = currNode
  }

  pop() {
    const currNode = this.top
    this.top = currNode.next
    return currNode
  }

  peek() {
    return this.top
  }

  isEmpty() {
    if (this.peek() === null) {
      return true
    } else {
      return false
    }
  }
}

exports.Stack = Stack

File: Utilities/Stack.rb
class Stack
    attr_accessor(:data, :top)
    def initialize()
      @data = []
      @top = nil
    end

    def push(value)
        self.data.push(value)
        self.top = value
    end

    def pop()
        value = self.data.pop()
        self.top = self.data[-1]
        return value
    end
end